src/contracts/core/DelegationManager.sol

```diff
@@ -113,18 +114,30 @@ contract DelegationManager is
      */
     function registerAsOperator(
         OperatorDetails calldata registeringOperatorDetails,
+        uint32 allocationDelay,
         string calldata metadataURI
     ) external {
         require(!isDelegated(msg.sender), "DelegationManager.registerAsOperator: caller is already actively delegated");
         _setOperatorDetails(msg.sender, registeringOperatorDetails);
         SignatureWithExpiry memory emptySignatureAndExpiry;
         // delegate from the operator to themselves
         _delegate(msg.sender, msg.sender, emptySignatureAndExpiry, bytes32(0));
+        // set the allocation delay
+        _initializeAllocationDelay(allocationDelay);
```

Comment:
this should call into the AVSD, so much cleaner for it to continue to live there

Reply:
requires a `onlyDelegationManager` check if its stored in the AVSD, seems kind of gross to me vs just doing the external call to the DM to read it when doing allocations

Reply:
let's debate!